rnd/autogpt_server/autogpt_server/blocks/discordblock.py

```diff
@@ -0,0 +1,196 @@
+import asyncio
+
+import aiohttp
+import discord
+
+from autogpt_server.data.block import Block, BlockOutput, BlockSchema
+from autogpt_server.data.model import BlockSecret, SecretField
+
+
+class DiscordReaderBlock(Block):
+    class Input(BlockSchema):
+        discord_bot_token: BlockSecret = SecretField(
+            key="discord_bot_token", description="Discord bot token"
+        )
+
+    class Output(BlockSchema):
+        message_content: str  # The content of the message received
+        channel_name: str  # The name of the channel the message was received from
+        username: str  # The username of the user who sent the message
```

Comment:
aside: can we use Field() to have these populate with descriptions on the frontend?